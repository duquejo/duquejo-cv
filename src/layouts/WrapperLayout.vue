<template>
  <div class="animated-container">
    <article ref="top">
      <slot />
    </article>
    <FooterComponent :social-links="SOCIAL_LINKS" />
  </div>
</template>

<script setup lang="ts">
import { onMounted, ref, defineAsyncComponent } from 'vue';

import SOCIAL_LINKS from '@/shared/data/social.json';

const FooterComponent = defineAsyncComponent(() => import('@/components/footer/FooterComponent.vue'));

const top = ref<null | HTMLDivElement>(null);

onMounted(() => {
  if (top.value) {
    top.value!.scrollIntoView({ behavior: 'instant' });
  }
});
</script>