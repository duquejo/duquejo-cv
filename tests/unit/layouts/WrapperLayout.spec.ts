import { shallowMount } from "@vue/test-utils";
import { describe, expect, it, vi } from "vitest";

import WrapperLayout from '../../../src/layouts/WrapperLayout.vue';

describe('+ WrapperLayout tests', () => {

  const scrollIntoViewSpy = vi.fn();
  window.HTMLElement.prototype.scrollIntoView = scrollIntoViewSpy;

  it('Should match the snapshot', () => {
    const wrapper = shallowMount(WrapperLayout);

    expect(wrapper.html()).toMatchSnapshot();

    expect(scrollIntoViewSpy).toHaveBeenCalled();
    expect(scrollIntoViewSpy).toHaveBeenCalledWith({
      behavior: 'instant'
    });
  });
});