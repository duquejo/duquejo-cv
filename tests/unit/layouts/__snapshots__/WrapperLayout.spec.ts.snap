// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`+ WrapperLayout tests > Should match the snapshot 1`] = `
"<div class="animated-container">
  <article class="mt-5 lg:mt-0"></article>
  <footer-component-stub social-links="[object Object],[object Object]"></footer-component-stub>
</div>"
`;
